Add an autogen.sh script to run all of the autotools in the right order
to regenerate the build machinery.  Needed since we have patches that
modify the source for Autoconf.

Debian-specific.

Index: b/autogen.sh
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ b/autogen.sh	2010-07-12 12:07:21.000000000 +0200
@@ -0,0 +1,29 @@
+#!/bin/sh
+#
+# autogen.sh
+
+LIBTOOL=`which libtoolize`
+AUTOCONF=`which autoconf`
+ACLOCAL=`which aclocal`
+
+if [ ! -x $LIBTOOL ]; then
+  echo "libtool doesn't exist! Please install libtool"
+  exit 1
+fi
+
+if [ ! -x $ACLOCAL ]; then
+  echo "aclocal doesn't exist! Please install automake"
+  exit 1
+fi
+
+if [ ! -x $AUTOCONF ]; then
+  echo "autoconf doesn't exist! Please install autoconf"
+  exit 1
+fi
+
+$LIBTOOL --copy --force
+$ACLOCAL
+$AUTOCONF
+
+exit 0
+
